File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.cpp
class Solution {
    vector<int> mem;
    int dp(vector<int>& cost, int i) {
        if (i >= cost.size())
            return 0;

        int& res = mem[i];
        if (res != -1)
            return res;

        return res = cost[i] + std::min(dp(cost, i + 1) , dp(cost, i + 2));  
    }
public:
    int minCostClimbingStairs(vector<int>& cost) {
        mem = vector<int>(cost.size() + 1, -1);
        return std::min(dp(cost, 0), dp(cost, 1));
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left = 0
        let right = heights.length - 1 

        let res = Number.MIN_SAFE_INTEGER


        while (left < right) {
            res = Math.max(res, Math.min(heights[left], heights[right]) * (right - left))

            if (heights[right] > heights[left])
                left++
            else
                right--    
        }

        return res
    }
}
